src/components/menu-surface/menu-surface.scss

```diff
@@ -10,7 +10,17 @@
 @include menu.core-styles;
 
 .mdc-menu-surface {
-    width: var(--menu-surface-width, auto);
     max-height: 100%;
     position: relative;
+    --mdc-menu-max-width: var(
+        --menu-surface-width,
+        min(calc(100vw - 2rem), 20rem)
+    );
+}
+
+:host(limel-menu-surface.has-grid-layout) {
+    .mdc-menu-surface {
+        width: var(--menu-surface-width, min(calc(100vw - 2rem), 40rem));
```

Comment:
Hmm… above we use `--mdc-menu-max-width`, but here we actually set the `width`, not the `max-width`. How come?